test/compress/issue-1443.js

```diff
@@ -0,0 +1,53 @@
+unsafe_undefined: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {}
+    input: {
+        function f(undefined) {
+            var a;
+            return function() {
+                console.log(typeof a == "undefined");
+            }
+        }
+    }
+    expect: {
+        function f(n) {
+            var o;
+            return function() {
+                console.log(n===o);
+            }
+        }
+    }
+}
+
+keep_fnames: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {
+        keep_fnames: true
+    }
+    input: {
+        function f(undefined) {
+            return function() {
+                function n(a) {
+                    return a * a;
+                }
+                return typeof n == "undefined";
+            };
+        }
+    }
+    expect: {
+        function f(r) {
+            return function() {
+                function n(n) {
+                    return n * n;
+                }
+                return r === n;
```

Comment:
(I'm little confused by your comment, so please bear with me...)

I added this test to ensure my removal of `ref.refernce();` call within `compress.js` would not cause #1431 to resurface in the form of this `unsafe` optimisation on `undefined`.

The reason why I use the roundabout way of `typeof n == "undefined"` instead of plain `undefined` is to make sure I end up with `AST_Undefined` instead of the local variable by that same name (which can be verified by running the code snippet against `master` and see the program crash).

Not sure if I've answered all/any of your questions? :sweat_smile: